torchmultimodal/modules/encoders/clip_vit_encoder.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn, Tensor
+
+# Taken from original clip implementation https://github.com/openai/CLIP/blob/main/clip/model.py#L167
+# TODO: unify with the implementation in text encoder
+def quick_gelu(x):
+    return x * torch.sigmoid(1.702 * x)
+
+
+class CLIPViTEncoder(nn.Module):
+    """
+    Vision transformer encoder for CLIP.
+
+    Args:
+        embedding_dim (int): Embedding dimension for text and image encoders projections.
+        patch_size (int): The dimension of each patch
+        image_size(int): The size (width==height) of input image
+        width (int): Dimensionality of the encoder layers and the pooler layer
+        heads (int): Number of attention heads for each attention layer in the Transformer encoder
+        layers (int): Number of hidden layers in the Transformer encoder
+
+    Inputs:
+        x (Tensor): image tensor with dimensions B x C(3) x image_size x image_size
```

Comment:
would it be worth having our own VisionTransformer? This is the third time we've had to customize the torchvision one (ALBEF changes one layer, FLAVA has another one from scratch)

Reply:
the clip vision transformer is a customized one with extra ln. I thought you were plannig to unify flava and albef one anyways, though i dont think this one needs to use it